tool/osl_pluto.c

```diff
@@ -1234,17 +1234,28 @@ static void compute_deps_osl(osl_scop_p scop, PlutoProg *prog,
       isl_map *schedule_i;
       isl_union_map *read_i;
       isl_union_map *write_i;
+
       char name[20];
 
+      int nb_parameters = 0;
+      int nb_iterators = 0;
+      int nb_scattdims = 0;
+      int nb_localdims = 0;
+      int array_id = 0;
+
       snprintf(name, sizeof(name), "S_%d", i);
 
-      int niter = osl_statement_get_nb_iterators(stmt);
-      space = isl_space_set_alloc(ctx, scop->context->nb_parameters, niter);
+      osl_relation_get_attributes(stmt->scattering, &nb_parameters,
+                                  &nb_iterators, &nb_scattdims, &nb_localdims,
+                                  &array_id);
+
+      space =
+          isl_space_set_alloc(ctx, scop->context->nb_parameters, nb_scattdims);
```

Comment:
Should it be `nb_iterators` here?

Reply:
`space` is being used for the `domain` below. So its dimensionality should be `nb_iterators`? It was `niter` earlier.